CDCgov/prime-reportstream
# CA receiver transforms

This PR updates CA receiver transforms to transform dates to CA's timezone. It also adds a Simple Report sender transforms to subtract 15 minutes from Specimen collected, Diagnostic Report issued and effective. Simple Report subtracts 15 minutes from dates being sent to us through the Covid pipeline. CA wants Simple Report to stop doing that for them, but Simple Report can't just turn it off for every message we receive. They will stop doing this for reports sent through the UP, but we will have to continue doing this in the UP to keep dates consistent across the pipelines. We will turn this off once we start routing the Covid results through the UP.

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Check out and build the branch
2. run `./gradlew reloadSettings`
3. Send a test report using the attached Postman collection
[Simple Report Copy.postman_collection.txt](https://github.com/CDCgov/prime-reportstream/files/11736647/Simple.Report.Copy.postman_collection.txt)
4. Verify that the request was accepted
5. Open the generated HL7 file in the local sftp folder.
6. Verify that dates have been converted to PST and that 15 minutes have been subtracted from the dates in `Bundle.entry.resource.ofType(DiagnosticReport).issued | effective`, and mapped to OBR-7, OBR-22, SPM-17, SPM-8 

## Changes
- Updated CA receiver transform
- Added Simple Report sender transform
-  Updated integration tests

## Checklist

### Testing

### Process

## Linked Issues
- #9923

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
